### PR TITLE
(TWILL-248) Upgrade to use Netty-4.1

### PR DESCRIPTION
**Pull request by chtyim**
- Also enable ResourceReportClient to use HTTP compression

### CODE REVIEW

File: twill-yarn/src/main/java/org/apache/twill/yarn/ResourceReportClient.java [hunk 1]
```diff
@@ -52,12 +54,16 @@
   public ResourceReport get() {
     for (URL url : resourceUrls) {
       try {
-        Reader reader = new BufferedReader(new InputStreamReader(url.openStream(), Charsets.UTF_8));
-        try {
+        HttpURLConnection urlConn = (HttpURLConnection) url.openConnection();
```
Comment (hsaputra) [inline twill-yarn/src/main/java/org/apache/twill/yarn/ResourceReportClient.java]: Is this change related to Netty upgrade?

**Comment by hsaputra** [comment]
One question about change form using BufferedReader,  but looks good.

If passes tests +1

File: twill-yarn/src/main/java/org/apache/twill/yarn/ResourceReportClient.java (hunk 1 above)
Comment (chtyim) [reply twill-yarn/src/main/java/org/apache/twill/yarn/ResourceReportClient.java]: Not entirely. This is to enable compression when fetching the report.
